overlord/snapstate/handlers.go

```diff
@@ -1956,39 +1956,44 @@ func nearlyReady(me string, change *state.Change) bool {
 	return true
 }
 
-// laneSnapSetups returns the instance names from the first SnapSetup in each
+// laneSnaps returns the instance names from the first SnapSetup in each
 // non-zero lane that is Done
-func laneSnaps(change *state.Change) []string {
+func laneSnaps(tid string, tasks []*state.Task) []string {
+	laneSnaps := map[int]string{}
 	// change.Tasks() preserves the order tasks were added, otherwise it all falls apart
-	tasks := change.Tasks()
-	sadLanes := map[int]bool{}
 	for _, task := range tasks {
-		if task.Status() == state.DoneStatus {
-			continue
+		if task.ID() == tid {
```

Comment:
this and the one after probably merit a comment on the fact they are based on change.Tasks()  returning tasks in order of addition to the change, that's also why my comment about doing change.Tasks() here and not in the caller (the caller could send in shuffled tasks for all we know right here)